### PR TITLE
fix gun encumbrance ui

### LINKED ISSUE #46391: Wearable guns don't display encumbrance
# Describe the bug

# Steps To Reproduce
Load the attached game and select the worn gun in the inventory.

# Expected behavior
Numbers showing up.

# Screenshots

# Versions and configuration
- OS: Linux
    - OS Version: <unknown>
- Game Version: 0.E-8531-gea3e6a3837-dirty [64-bit]
- Graphics Version: Curses
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Alternative Map Key [alt_map_key],
    No Rail Stations [No_Rail_Stations]
]

# Additional context
[2020.12.19.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/5750488/2020.12.19.zip)

### PR DESCRIPTION
**Pull request by Jamuro-g**
#### Summary
Bugfixes "Wearable guns now display their encumbrance"

#### Purpose of change
Guns with mods that made them wearable didn't show their encumbrance in the examine and modify ui.
This should fix that

Fixes #46391

#### Describe the solution

Added a check for guns in item::armor_info(), that cycles through mods and identifies the location via assigned flags

#### Describe alternatives you've considered

Originally i wanted to put this check inside find_armor_data() but sadly i didn't find a decent way to retun the added data as a const islot_armor.

#### Testing

- Spawn in a m4a1 and a shoulder strap
- unload the gun and attach the strap
- the modification ui should now show the encumbrance 
- examine the modified gun

same for the belt clip

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -2867,6 +2867,33 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
...
             }
+        } else if( is_gun() ) {
+
+            std::string bp_name;
+            bodypart_id bp_id;
+            for( const item *mod : gunmods() ) {
+                if( has_flag( flag_IS_ARMOR ) ) {
+                    //todo: add further checks for specific location flags whenever they get added
+                    if( mod->has_flag( flag_BELTED ) ) {
+                        bp_name = "Torso";
```
Comment (BrettDong) [inline src/item.cpp:2878]:
```suggestion
                        bp_name = _( "Torso" );
```

File: src/item.cpp [hunk 2]
```diff
@@ -2867,6 +2867,33 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
...
+                        bp_id = body_part_torso.id();
+                    } else {
+                        //in case no location specific flag is set (like with belt clips)
+                        bp_name = "Torso";
```
Comment (BrettDong) [inline src/item.cpp:2882]:
```suggestion
                        bp_name = _( "Torso" );
```

File: src/item.cpp [hunk 3]
```diff
@@ -2867,6 +2867,33 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
...
                 }
             }
+        } else if( is_gun() ) {
+
+            std::string bp_name;
+            bodypart_id bp_id;
+            for( const item *mod : gunmods() ) {
+                if( has_flag( flag_IS_ARMOR ) ) {
+                    //todo: add further checks for specific location flags whenever they get added
+                    if( mod->has_flag( flag_BELTED ) ) {
```
Comment (BrettDong) [inline src/item.cpp:2877]: The code blocks in `if` and in `else` are the same??

Comment (Jamuro-g) [reply src/item.cpp:2877]: it is redundant yeah ... i was mostly there to remind me to check for further location flags that gunmods might have. i ll remove it.

File: src/item.cpp [hunk 4]
```diff
@@ -2867,6 +2867,27 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
                     }
                 }
             }
+        } else if( is_gun() ) {
+
+            std::string bp_name;
+            bodypart_id bp_id;
+            for( const item *mod : gunmods() ) {
+                if( has_flag( flag_IS_ARMOR ) ) {
```
Comment (BrettDong) [inline src/item.cpp:2875]: Are you checking the gun item itself, or the gunmod item?

Comment (Jamuro-g) [reply src/item.cpp:2875]: oh wow, yeah the for loop should have been inside that if (has_flag already checks for inherited flags from gunmods)

i probably should remove the for loop ... it isn't needed, at least until we get gunmods that require specific flag checks to be associated with a body part other than torso.

File: src/item.cpp [hunk 5]
```diff
@@ -2868,16 +2868,14 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
...
-                    bp_name = _( "Torso" );
-                    bp_id = body_part_torso.id();
-                }
+            if( has_flag( flag_IS_ARMOR ) ) {
+                //right now all eligible gunmods (shoulder_strap, belt_clip use the torso)
+                bp_name = _( "Torso" );
+                bp_id = body_part_torso.id();
             }
+
             if( !bp_name.empty() ) {
```
Comment (BrettDong) [inline src/item.cpp:2879]: If so then you can just `if( has_flag( flag_IS_ARMOR ) ) {` and get rid of variable `bp_name`.

Comment (Jamuro-g) [reply src/item.cpp]: ok

**Review by BrettDong — APPROVED** [review]

**Pull request merged**